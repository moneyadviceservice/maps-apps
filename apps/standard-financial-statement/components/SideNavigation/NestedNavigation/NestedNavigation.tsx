import { Fragment } from 'react';

import { SideNavigationItemGroup } from 'types/@adobe/components';
import { v4 as uuidv4 } from 'uuid';

import { Heading } from '@maps-react/common/components/Heading';

import { NavigationItem } from '../NavigationItem';

type NavigationGroupsProps = {
  language: string;
  sideNavigationGroups: SideNavigationItemGroup[];
  slug: string[];
};

export const NestedNavigation = ({
  language,
  sideNavigationGroups,
  slug,
}: NavigationGroupsProps) => {
  return (
    <nav className="bg-white min-w-[300px] lg:max-w-[300px] lg:sticky lg:top-4 lg:overflow-y-auto">
      <div className="lg:border-1 lg:border-slate-400 lg:border-r-r lg:rounded-bl-[24px] pb-4 lg:pb-0 mb-8">
        {sideNavigationGroups.map((item, index) => {
          const isLastGroup = index === sideNavigationGroups.length - 1;

          return (
            <Fragment key={uuidv4()}>
              {item?.title && (
                <Heading
                  level="h5"
                  className="pt-6 pb-3 mx-5 border-b-1 border-b-slate-400"
                >
                  {item?.title}
                </Heading>
              )}

              {item?.childLinks?.length > 0 && (
                <ul>
                  {item?.childLinks.map((link, itemIndex) => (
                    <NavigationItem
                      isGroup={true}
                      key={uuidv4()}
                      lang={language}
                      linkItem={link}
                      isLastLink={
                        isLastGroup && itemIndex === item?.childLinks.length - 1
                      }
                      isActive={
                        slug && slug[0] === (link as { linkTo: string }).linkTo
                      }
                    />
                  ))}
                </ul>
              )}
            </Fragment>
          );
        })}
      </div>
    </nav>
  );
};
