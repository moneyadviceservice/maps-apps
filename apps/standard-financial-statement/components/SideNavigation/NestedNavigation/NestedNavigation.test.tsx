import { render } from '@testing-library/react';

import { NestedNavigation } from './NestedNavigation';

jest.mock('next/router', () => ({
  useRouter: jest.fn().mockReturnValue({
    query: {},
    asPath: '/',
  }),
}));

export const sideNavigationGroups = [
  {
    title: null,
    childLinks: [
      {
        text: 'Use the SFS',
        linkTo: '/use-the-sfs',
        description: null,
      },
    ],
  },
  {
    title: 'Spending Guidelines',
    childLinks: [
      {
        text: 'Spending Guidelines 2024/25',
        linkTo: '/use-the-sfs/spending-guidelines',
        description: null,
      },
      {
        text: 'Commentary for 2024/25',
        linkTo: '/use-the-sfs/commentary-for',
        description: null,
      },
    ],
  },
];

describe('NestedNavigation component', () => {
  it('renders nested nav correctly', () => {
    const { container } = render(
      <NestedNavigation
        language="en"
        sideNavigationGroups={sideNavigationGroups}
        slug={['/use-the-nested-sfs']}
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
