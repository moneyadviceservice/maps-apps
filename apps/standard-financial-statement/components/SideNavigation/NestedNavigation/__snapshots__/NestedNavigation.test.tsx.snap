// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NestedNavigation component renders nested nav correctly 1`] = `
<div>
  <nav
    class="bg-white min-w-[300px] lg:max-w-[300px] lg:sticky lg:top-4 lg:overflow-y-auto"
  >
    <div
      class="lg:border-1 lg:border-slate-400 lg:border-r-r lg:rounded-bl-[24px] pb-4 lg:pb-0 mb-8"
    >
      <ul>
        <li>
          <a
            class="gap-2 focus:text-gray-800 focus-within:outline-0 focus:no-underline cursor-pointer active:text-gray-800 active:underline active:bg-transparent active:shadow-none text-sm text-magenta-700 visited:text-magenta-700 px-4 lg:px-6 py-3 flex items-center after:content-[""] after:absolute after:bottom-[-1px] after:left-4 after:right-4 lg:after:left-6 lg:after:right-6 after:h-[1px] after:bg-slate-400 no-underline hover:underline hover:bg-gray-300 focus:bg-yellow-200 relative focus:shadow-transparent z-1"
            href="/en/use-the-sfs"
          >
            <span
              class="max-w-[90%]"
            >
              Use the SFS
            </span>
            <chevron.svg
              class="ml-auto"
            />
          </a>
        </li>
      </ul>
      <h5
        class="text-lg md:text-xl text-gray-800 font-bold pt-6 pb-3 mx-5 border-b-1 border-b-slate-400"
      >
        Spending Guidelines
      </h5>
      <ul>
        <li>
          <a
            class="gap-2 focus:text-gray-800 focus-within:outline-0 focus:no-underline cursor-pointer active:text-gray-800 active:underline active:bg-transparent active:shadow-none text-sm text-magenta-700 visited:text-magenta-700 px-4 lg:px-6 py-3 flex items-center after:content-[""] after:absolute after:bottom-[-1px] after:left-4 after:right-4 lg:after:left-6 lg:after:right-6 after:h-[1px] after:bg-slate-400 no-underline hover:underline hover:bg-gray-300 focus:bg-yellow-200 relative focus:shadow-transparent z-1"
            href="/en/use-the-sfs/spending-guidelines"
          >
            <span
              class="max-w-[90%]"
            >
              Spending Guidelines 2024/25
            </span>
            <chevron.svg
              class="ml-auto"
            />
          </a>
        </li>
        <li>
          <a
            class="gap-2 focus:text-gray-800 focus-within:outline-0 focus:no-underline cursor-pointer active:text-gray-800 active:underline active:bg-transparent active:shadow-none text-sm text-magenta-700 visited:text-magenta-700 px-4 lg:px-6 py-3 flex items-center after:absolute after:bottom-[-1px] after:left-4 after:right-4 lg:after:left-6 lg:after:right-6 after:h-[1px] after:bg-slate-400 no-underline hover:underline hover:bg-gray-300 focus:bg-yellow-200 relative focus:shadow-transparent z-1 lg:rounded-bl-[23px] after:content-none"
            href="/en/use-the-sfs/commentary-for"
          >
            <span
              class="max-w-[90%]"
            >
              Commentary for 2024/25
            </span>
            <chevron.svg
              class="ml-auto"
            />
          </a>
        </li>
      </ul>
    </div>
  </nav>
</div>
`;
